components/single_image_view/single_image_view.tsx

```diff
@@ -16,25 +17,35 @@ import ViewImageModal from 'components/view_image';
 
 const PREVIEW_IMAGE_MIN_DIMENSION = 50;
 
-export default class SingleImageView extends React.PureComponent {
-    static propTypes = {
-        postId: PropTypes.string.isRequired,
-        fileInfo: PropTypes.object.isRequired,
-        isRhsOpen: PropTypes.bool.isRequired,
-        compactDisplay: PropTypes.bool,
-        isEmbedVisible: PropTypes.bool,
-        actions: PropTypes.shape({
-            toggleEmbedVisibility: PropTypes.func.isRequired,
-        }).isRequired,
+type Props = {
+    post: Post;
+    fileInfo: FileInfo;
+    isRhsOpen: boolean;
+    compactDisplay?: boolean;
+    isEmbedVisible?: boolean;
+    actions: {
+        toggleEmbedVisibility: (postId: string) => void;
     };
+}
+
+type State = {
+    loaded: boolean;
+    showPreviewModal: boolean;
+    dimensions: {
+        width: number;
+        height: number;
+    },
+}
 
+export default class SingleImageView extends React.PureComponent<Props, State> {
+    private mounted: boolean;
     static defaultProps = {
-        fileInfo: {},
```

Comment:
Are we sure the default value won't be used?

Reply:
Hi @Tak-Iwamoto, thanks for your contribution.  I would like to approve this PR but need the above question addressed first.  Thanks!